sympy/physics/units/quantities.py

```diff
@@ -117,9 +116,15 @@ def _collect_factor_and_dimension(expr):
             factor, dim = Quantity._collect_factor_and_dimension(expr.base)
             return factor ** expr.exp, dim ** expr.exp
         elif isinstance(expr, Add):
-            raise NotImplementedError
+            factor, dim = Quantity._collect_factor_and_dimension(expr.args[0])
```

Comment:
Are you doing `args[0]` to pull off constant factors? If so, you can use `as_coeff_add()`, which gives you `0` for the coeff if there is no rational/float constant factor.